packages/SwingSet/src/kernel/kernel.js

```diff
@@ -808,6 +808,10 @@ export default function buildKernel(kernelEndowments) {
       const vatKeeper = kernelKeeper.allocateVatKeeperIfNeeded(vatID);
       if (!vatKeeper.isDead()) {
         vatKeeper.markAsDead();
+        const err = makeError('vat terminated');
+        for (const kpid of kernelKeeper.findPromisesFromDeadVat(vatID)) {
+          deliverToError(kpid, err, vatID);
```

Comment:
I'm thinking that maybe this function should now be called `resolveToError`, since it's not dealing with messages anymore, only an unresolved promise. Maybe something for a later PR.

Reply:
It's a trivial change.  I'll drop it in.